tests/scanner/audit/external_project_access_engine_test.py

```diff
@@ -0,0 +1,220 @@
+        """Test that a RuleBook is built correctly with a yaml file."""
+        rules_local_path = get_datafile_path(__file__,
+            'external_project_access_test_rules_2.yaml')
+        rules_engine = epare.ExternalProjectAccessRulesEngine(rules_file_path=rules_local_path)
+        with self.assertRaises(audit_errors.InvalidRulesSchemaError):
+            rules_engine.build_rule_book(self.inventory_config)
+
+    def test_no_viloations(self):
+        """Test that a RuleBook is built correctly with a yaml file."""
+        all_violations = []
+        rules_local_path = get_datafile_path(__file__,
+            'external_project_access_test_rules_1.yaml')
+        rules_engine = epare.ExternalProjectAccessRulesEngine(rules_file_path=rules_local_path)
+        rules_engine.build_rule_book(self.inventory_config)
+        for user, ancestry in self.TEST_ANCESTRIES.iteritems():
+            violations = rules_engine.find_policy_violations(user, ancestry)
+            all_violations.extend(violations)
+        self.assertEqual(len(all_violations), 0)
+
+    def test_no_viloations_no_rules(self):
+        """Test that a RuleBook is built correctly with a yaml file."""
+        all_violations = []
+        rules_local_path = get_datafile_path(__file__,
+            'external_project_access_test_rules_0.yaml')
+        rules_engine = epare.ExternalProjectAccessRulesEngine(rules_file_path=rules_local_path)
+        rules_engine.build_rule_book(self.inventory_config)
+        for user, ancestry in self.TEST_ANCESTRIES_SIMPLE.iteritems():
+            violations = rules_engine.find_policy_violations(user, ancestry)
+            all_violations.extend(violations)
+        self.assertEqual(len(all_violations), 0)
+
+    def test_yes_viloations(self):
+        """Test that a RuleBook is built correctly with a yaml file."""
+        all_violations = []
+        rules_local_path = get_datafile_path(__file__,
+            'external_project_access_test_rules_1.yaml')
+        rules_engine = epare.ExternalProjectAccessRulesEngine(rules_file_path=rules_local_path)
+        rules_engine.build_rule_book(self.inventory_config)
+        for user, ancestry in self.TEST_ANCESTRIES_VIOLATIONS.iteritems():
+            violations = rules_engine.find_policy_violations(user, ancestry)
+            all_violations.extend(violations)
+        self.assertEqual(len(all_violations), 2)
+
+
+class ExternalProjectAccessRuleBookTest(ForsetiTestCase):
+    """Tests for the ExternalProjectAccessRuleBook."""
+
+    TEST_GOOD_RULE = dict(name='default', allowed_ancestors=['organizations/7890'])
+    TEST_BAD_RULE = dict(name='default', allowed_ancestors=['policy/12345'])
+    TEST_RULE_DEFS = dict(rules=[TEST_GOOD_RULE])
+
+    TEST_ANCESTORS = [Project("123"), Folder("456"), Organization("7890")]
+    TEST_BAD_ANC = [Project("123"), Folder("456"), Organization("ABC")]
+
+    def setUp(self):
+        """Set up."""
+        self.rule_index = 0
+        self.epare = epare
```

Comment:
When I read this, my first question is: what is `epare`?  Do you want to just call this, `rule_engine`?

Reply:
it stands for **e**xternal **p**roject **a**ccess **r**ules **e**ngine.  But that's obtuse.  ```rules_engine``` is better